[quant][graphmode] Default to non-inplace in graph mode quantization API

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32379 [WIP][quant][graphmode] FoldConvBatchNorm2d support shared ClassTypes
* #32380 [quant][graphmode][refactor] Better API for fold_convbn
* #32374 [refactor] Adding FoldConvBatchNorm2dHelper
* #32303 [quant][graphmode] Call _jit_pass_dedup_module_ueses in quantize_script
* #32302 [quant] Re-enable fold_convbn in quantize_script
* #32206 [quant] Re-enable test_nested that has different qconfig for shared ClassType
* #32205 [quant][graphmode] Support quantizing shared ClassType with different qconfigs
* **#32204 [quant][graphmode] Default to non-inplace in graph mode quantization API**
* #32203 Adding QConfigTypePtrMap
* #32202 [refactor] Move some of the helper functions for public use
* #31408 [quant][graphmode] `insert_quant_dequant` pass support shared class types

Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19508030](https://our.internmc.facebook.com/intern/diff/D19508030)

jerryzh168 committed:
[quant][graphmode] Default to non-inplace in graph mode quantization API

Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 6c1c6bd1:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=6c1c6bd1558c873daade9745b8d378fe4d358ed3).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248388) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86222654) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86222654))
<details>
<summary>
<code>Jan 15 02:39:47 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff</code>
</summary>

```
Jan 15 02:39:47 + XDIR=/var/lib/jenkins/workspace/xla/scripts/.. 
Jan 15 02:39:47 + PTDIR=/var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 15 02:39:47 + python /var/lib/jenkins/workspace/xla/scripts/cond_patch.py /var/lib/jenkins/workspace/xla/scripts/../torch_patches /var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 15 02:39:47 1 out of 1 hunk FAILED 
Jan 15 02:39:47 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 15 02:39:47 Traceback (most recent call last): 
Jan 15 02:39:47   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module> 
Jan 15 02:39:47     patch_repo(args) 
Jan 15 02:39:47   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo 
Jan 15 02:39:47     raise RuntimeError('Failed to apply patch: {}'.format(ppath)) 
Jan 15 02:39:47 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 15 02:39:47 + cleanup 
Jan 15 02:39:47 + retcode=1 
Jan 15 02:39:47 + set +x 
Jan 15 02:39:47 =================== sccache compilation log =================== 
Jan 15 02:39:47 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 15 02:39:47 Compile requests                 1 
Jan 15 02:39:47 Compile requests executed        0 
Jan 15 02:39:47 Cache hits                       0 
Jan 15 02:39:47 Cache misses                     0 
Jan 15 02:39:47 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jerryzh168 committed:
Update on "[quant][graphmode] Default to non-inplace in graph mode quantization API"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Default to non-inplace in graph mode quantization API"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Default to non-inplace in graph mode quantization API"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Default to non-inplace in graph mode quantization API"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Default to non-inplace in graph mode quantization API"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Default to non-inplace in graph mode quantization API"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/quantization/_quantize_script.py:
```diff
@@ -69,17 +70,17 @@ def prepare_script(model, qconfig_dict, inplace=False):
     _check_is_script_module(model)
     if not inplace:
         model = model.copy()
```
That copy above seems redundant then.

jerryzh168 replied:
yeah, I'll do another refactor to remove `inplace` argument

ZolotukhinM commented on torch/quantization/_quantize_script.py:
```diff
@@ -103,13 +104,7 @@ def quantize_script(model, qconfig_dict, run_fn, run_args, inplace=False):
     # JIT
     # torch._C._jit_pass_fold_convbn(model._c)
-    prepare_script(model, scripted_qconfig_dict, True)
+    model = prepare_script(model, scripted_qconfig_dict, True)
```
Should the last argument also be `False`?

jerryzh168 replied:
I'll do another refactor to remove this argument

ZolotukhinM commented on torch/quantization/_quantize_script.py:
```diff
@@ -103,13 +104,7 @@ def quantize_script(model, qconfig_dict, run_fn, run_args, inplace=False):
-    # [TODO] This will be fixed later when we figure out
-    # how to properly mutate types
-    model = convert_script(model, False)
+    model = convert_script(model, True)
```
Here as well.

ZolotukhinM reviewed (approved):
👍 
Some comments are inline.

jerryzh168 committed:
Update on "[quant][graphmode] Default to non-inplace in graph mode quantization API"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Default to non-inplace in graph mode quantization API"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Default to non-inplace in graph mode quantization API"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
